gdx/src/com/badlogic/gdx/utils/ObjectMap.java

```diff
@@ -1,846 +1,782 @@
+	 * <br>
+	 * This can be overridden by subclasses, which you may want to do if your key type needs special consideration for
+	 * its hash (such as if you can't modify or extend a particular class that has an incorrect hashCode()). Subclasses
+	 * that don't need the collision decrease of Fibonacci Hashing (assuming the key class has a good hashCode()) may do
+	 * fine with a simple implementation:
+	 * {@code return (item.hashCode() & mask);}
+	 *
+	 * @param item a key that this method will hash, by default by calling {@link Object#hashCode()} on it; non-null
+	 * @return an int between 0 and {@link #mask}, both inclusive
+	 */
+	protected int place (final K item) {
+		// shift is always greater than 32, less than 64
+		return (int)(item.hashCode() * 0x9E3779B97F4A7C15L >>> shift);
+	}
+
+	private int locateKey (final K key) {
+		return locateKey(key, place(key));
+	}
+
+	/**
+	 * Given a key and its initial placement to try in an array, this finds the actual location of the key in the array
+	 * if it is present, or -1 if the key is not present. This can be overridden if a subclass needs to compare for
+	 * equality differently than just by calling {@link Object#equals(Object)}, but only within the same package.
+	 *
+	 * @param key       a K key that will be checked for equality if a similar-seeming key is found
+	 * @param placement as calculated by {@link #place(Object)}, almost always with {@code place(key)}
+	 * @return the location in the key array of key, if found, or -1 if it was not found.
+	 */
+	int locateKey (final K key, final int placement) {
+		for (int i = placement; ; i = i + 1 & mask) {
+			// empty space is available
+			if (keyTable[i] == null) {
+				return -1;
+			}
+			if (key.equals(keyTable[i])) {
+				return i;
+			}
+		}
+	}
+
+	/**
+	 * Returns the old value associated with the specified key, or null.
+	 */
+	public V put (K key, V value) {
+		if (key == null)
+			throw new IllegalArgumentException("key cannot be null.");
+		K[] keyTable = this.keyTable;
+		V[] valueTable = this.valueTable;
+		int b = place(key);
+		int loc = locateKey(key, b);
+		// an identical key already exists
+		if (loc != -1) {
+			V tv = valueTable[loc];
+			valueTable[loc] = value;
+			return tv;
+		}
+		for (int i = b; ; i = (i + 1) & mask) {
+			// space is available so we insert and break (resize is later)
+			if (keyTable[i] == null) {
+				keyTable[i] = key;
+				valueTable[i] = value;
+				break;
+			}
+		}
+		if (++size >= threshold) {
+			resize(keyTable.length << 1);
+		}
+		return null;
+	}
+
+	//	public void putAll (ObjectMap<K, V> map) {
+//		ensureCapacity(map.size);
+//		for (Entry<K, V> entry : map)
+//			put(entry.key, entry.value);
+//	}
+	public void putAll (ObjectMap<K, V> map) {
+		ensureCapacity(map.size);
+		final K[] keyTable = map.keyTable;
+		final V[] valueTable = map.valueTable;
+		K k;
+		for (int i = 0, n = keyTable.length; i < n; i++) {
+			if ((k = keyTable[i]) != null)
```

Comment:
For readability, I'd prefer assignment on a separate line, then checking `k != null`.

Reply:
This shows up in a lot of places, I think. I'll try a mass find/replace and see how it goes.